### DISCUSSION TITLE
How to mask some JSON response fields?

### DISCUSSION THREAD

**Discussion by vladnech** [ThreeMammals/Ocelot#2015] [Q&A] [answered]
Hi guys

I am not sure where to ask the question for the Ocelot team. 

We have to mask some PII data coming from our API Services to vue.js client (public website). 
For example, Mobile: 0123456789 to return as xxxx6789.

At this stage, before creating specific methods for this (reluctant of doing it because API can be used by other clients, where the mask is not going to be used), is there any way to do it on the Ocelot side? 

Thanks in advance for your help.

**Comment by raman-m** [comment, marked as answer]
Hi @vladnech !
You can transform data and convert JSON fields by the following:
- Use [Delegating Handlers](https://ocelot.readthedocs.io/en/latest/features/delegatinghandlers.html) to create transformed `HttpResponseMessage` object for selected routes with defined handler type
- Use [Middleware Injection](https://ocelot.readthedocs.io/en/latest/features/middlewareinjection.html) to develop one common middleware which function will be JSON data transformations **for all routes**.
- Use [Request Aggregation](https://ocelot.readthedocs.io/en/latest/features/requestaggregation.html) if you would like to aggregate JSON data of multiple routes.

The most relevant is delegating handler scenario. This will be a lite solution.

Hope it helps!
